### PR TITLE
Ga default campaign [MAILPOET-2398]

### PR DESCRIPTION
**Pull request by JanJakes**

### CODE REVIEW

File: assets/js/src/newsletters/send.jsx [hunk 1]
```diff
@@ -13,6 +13,14 @@ import Hooks from 'wp-js-hooks';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
 import ReactStringReplace from 'react-string-replace';
+import slugify from 'slugify';
+
+const generateGaTrackingCampaignName = (id, subject) => {
+  const name = slugify(subject, { lower: true })
+    .replace(/[^a-z0-9]+/g, '-')
```
[the remark this answers is not in the document]
Comment (JanJakes) [reply assets/js/src/newsletters/send.jsx]: @amine-mp I thought about that but it drops them completely without the `-` replacement 😞  So then `one+++two+three` becomes `onetwothree` while my method would make `one-two-three`. 

It's true that my method also converts i.e. group of Chinese symbols to `-` but I think most common are either subjects with a few special chars or smileys (the `-` is expected) or fully in an "unsupported" alphabet which would result in an empty string and therefore the default `newsletter` slug.

What do you think?

**Comment by codemonkey-jack** [comment]
@JanJakes This fails QA. If I'm reading the ticket correctly we aren't doing a dynamic update if the user updates the `subject line` on the last step page. 

However, on the last step page if I click "Go back to the design page" and edit the subject line there in the newsletter editor, the GA tracking is still my original subject line when I click through to the last step page. 

Shouldn't this be updated to the new subject line?

**Comment by JanJakes** [comment]
@codemonkey-jack I don't understand what you mean by `we aren't doing a dynamic update if the user updates the subject line on the last step page`. This is implemented:
> B- Dynamic update
> On the last step page, if the user rewrites the subject line and the GA tracking is still the default one, update it dynamically.

This:
> However, on the last step page if I click "Go back to the design page" and edit the subject line there in the newsletter editor, the GA tracking is still my original subject line when I click through to the last step page.

Makes sense, I'll have a look.

**Comment by codemonkey-jack** [comment]
@JanJakes 

>I don't understand what you mean by we aren't doing a dynamic update if the user updates the subject line on the last step page. This is implemented:

Ah, ok. This doesn't work either then. I was basing that assumption off Pavel's and Amine's comments on the Jira ticket where they both mention about the dynamic update being difficult. 

Anyway, as it is implemented it doesn't work either for me. 

>Makes sense, I'll have a look.

Thanks!

**Comment by JanJakes** [comment]
@codemonkey-jack I just tried and it works. It stops live-changing the value when you touch it manually, of course.

For the problem with going back to the design page, how important do you think it is? The behavior you're seeing is kind of expected and it would be rather hard to fix. Let me explain:
1. We prefill the value for GA tracking for user on the second page. Done only in JS (due to the live-change requirement).
2. When a user is happy with that form and saves it, the value is saved and we no longer touch it.
3. If you go back to the design page, you saved the newsletter and the value exists in DB. Even if it's default we do have to save it - we don't know if it's the last save before sending or not.
4. When you change subject on design page and then hit "save" (or autosave kicks in) we have no idea what was the original subject which means we don't know if GA tracking is default or not - it simply exists in DB which means the user saved it, being it default or not, we shouldn't really touch it.

It think this is really an edge case. If you have an idea how to solve it we could go for it but I don't really think it is worth it. We could for instance render a hidden `ga_campaign` field and change it when typing as we do in the second step. But that would mean duplicating all the logic in the editor which is not a React code and that just seems overly complex.

I'll push a rebase on master now but otherwise, there doesn't seem to anything to update for me.

**Comment by codemonkey-jack** [comment]
@JanJakes Hmm... The dynamic update on the last step works for me as well now. No idea why. 

For the other issue. I see your point and I don't think it's worth the time sink to do it. As it wasn't originally mentioned in the specs either, we'll leave this "as is" and I'll merge it. :)

**Pull request merged**